spec/API_specification/signatures/indexing_functions.py

```diff
@@ -0,0 +1,28 @@
+from ._types import Union, array
+
+def take(x: array, indices: Union[int, array], /, *, axis: int) -> array:
+    """
+    Returns elements of an array along an axis.
+
+    .. note::
+       Conceptually, the call ``take(x, indices, axis=3)`` is equivalent to ``x[:,:,:,indices,...]``; however, explicit indexing via arrays of indices is not currently supported in this specification due to concerns regarding ``__setitem__`` and array mutation semantics.
+
+    Parameters
+    ----------
+    x: array
+        input array.
+    indices: Union[int, array]
```

Comment:
Allowing a scalar is unusual. It's probably worth doing here because this function is a direct mapping from indexing syntax, where it is allowed. I wanted to point it out though, since we don't allow array-like's almost anywhere else.

Reply:
I'd also be okay with limiting `indices` to `array` input. This would simplify `take` behavior as spec'd. As currently spec'd, `take` has distinct behavior depending on whether `indices` is an `array` or an `integer`. Accordingly, given

```python
xp.take(x, 1)
```

you need to remember that this will result in a dimension reduction; whereas,

```python
xp.take(x, xp.array([1]))
```

does not result in a dimension reduction.

Furthermore, limiting to only array `indices` would also mean that there is only one way to do things. If a user wants to access a single axis, they should use explicit indexing

```python
x[:, :, 1, :]
```

Reply:
The proposed spec has been updated accordingly.